src/Resources/MenuItem.php

```diff
@@ -131,17 +132,22 @@ public function fields(Request $request)
             Text::make(trans('laravel-nova-menu::menu.external_link'), 'external_link')
                 ->help(trans('laravel-nova-menu::menu.must_start_with_http'))
                 ->nullable()
-                ->rules('max:191', function ($attribute, $value, $fail) {
+                ->rules('max:191', 'required_without:internal_link', function ($attribute, $value, $fail) {
                     if (!empty($value) && !Validator::make([$attribute => $value], [$attribute => 'url'])->passes()) {
                         return $fail(trans('laravel-nova-menu::errors.bad_format_external_link'));
                     }
                 })
                 ->hideFromIndex()
                 ->hideFromDetail(),
 
-            Select::make(trans('laravel-nova-menu::menu.internal_link'), 'internal_link')
+            Select2::make(trans('laravel-nova-menu::menu.internal_link'), 'internal_link')
                 ->options(MenuHelper::links())
                 ->rules('nullable', 'required_without:external_link', 'in:'.implode(',', array_keys(MenuHelper::links())))
+                ->configuration([
+                    'width' => '100%',
+                    'allowClear' => true,
+                    'multiple' => false,
+                ])
```

Comment:
Will be cool to set a threshold to trigger the serachbox to appear with
 `'minimumResultsForSearch' => $someInteger,`